### PR TITLE
Updated godoc comments.

### PR DESCRIPTION
**Pull request by cmars**
Just some minor godoc improvements, as recommended in http://blog.golang.org/godoc-documenting-go-code.

### CODE REVIEW

**Comment by ox** [comment]
This looks great, thanks! I'll let @vadimg give it a 1-over before pushing the button.

File: decimal.go [hunk 1]
```diff
@@ -279,20 +284,23 @@ func (d Decimal) Cmp(d2 Decimal) int {
 	return rd.value.Cmp(rd2.value)
 }
 
+// Equals returns whether d == d2 at the smallest scale among d and d2.
```
Comment (vadimg) [inline decimal.go]: This comment is confusing, just say that it returns whether the numbers represented by d and d2 are equal, regardless of their internal decimal representation.

File: decimal.go [hunk 2]
```diff
@@ -265,11 +269,12 @@ func (d Decimal) Div(d2 Decimal) Decimal {
 	return ret
 }
 
-// Cmp compares x and y and returns -1, 0 or 1
+// Cmp compares d and d2 and returns -1, 0 or 1.
+// The smallest scale of d and d2 are used in comparison.
```
Comment (vadimg) [inline decimal.go]: This comment is also confusing, just say it compares the numbers represented by the decimals, regardless of what their internal representation is.

**Comment by vadimg** [comment]
Looks good, thanks! Just a comment on two of your comments...

**Comment by cmars** [comment]
Updated, PTAL

**Comment by vadimg** [comment]
Thanks for adding those comments!

**Pull request merged**